### PR TITLE
Issue #14019: Resolve Pitest Supression For getCheckstyleModules Of ModuleReflectionUtil

### PR DESCRIPTION
**Pull request by piyush-1234**
Issue: #14019 

not ready for review

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java [hunk 1]
```diff
@@ -60,7 +60,6 @@ public static Set<Class<?>> getCheckstyleModules(
         return packages.stream()
                 .flatMap(pkg -> classPath.getTopLevelClasses(pkg).stream())
                 .map(ClassPath.ClassInfo::load)
-                .filter(ModuleReflectionUtil::isCheckstyleModule)
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java:63]: I don't see this as being a good change. We only allow checkstyle modules to be picked up. We don't want a bunch of random files to be scanned. It also breaks the javadoc and method name contract. We only pull in checkstyle modules.

**Review by rnveach — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java (hunk 1 above)
Comment (piyush-1234) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java:63]: yeah @rnveach , i was just trying with this change, whether i can find any regression test differences or not, i just made changes to see how it goes.

sure i do possible change if needed. once i see both regression report

Comment (rnveach) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java:63]: You can always put a PR into draft state, or make a statement in the first post, to show you are not ready for people to look at it.

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java:63]: this mutation most likely should be covered by specific junit, not BDD styled test (with Input) as we usually do.
We have IndentationCheck that has bunch of classes around it and they are not modules.

**Comment by piyush-1234** [comment]
hi @romani ,

i am working professional, having 2+ years of experience as software engineer.

i have started my contribution in Checkstyle from last October 2023 and I am still continuing with it.

i have one question.

am i eligible to take participation in GSOC 2024.

**Comment by romani** [comment]
> am i eligible to take participation in GSOC 2024.

yes, please finish this PR.
Activity in PR is most critical criteria for selection to GSOC.

in no activity in 2 weeks, we will close this PR.

**Pull request closed without merging**